guake/guake_app.py

```diff
@@ -1113,23 +1113,25 @@ def load_cwd_guake_yaml(self, vte) -> dict:
             return {}
 
         cwd = Path(vte.get_current_directory())
-        guake_yml = cwd.joinpath(".guake.yml")
-        content = {}
+        filename = str(cwd.joinpath(".guake.yml"))
 
-        if self._guake_yml_load_monotonic.get(guake_yml, 0.0) + 1.0 > pytime.monotonic():
-            content = self._guake_yml.get(guake_yml, {})
-        else:
-            try:
-                if guake_yml.is_file():
-                    with guake_yml.open(encoding="utf-8") as fd:
-                        content = yaml.safe_load(fd)
-                        self._guake_yml[guake_yml] = content
-                        self._guake_yml_load_monotonic[guake_yml] = pytime.monotonic()
-            except PermissionError:
-                log.debug("PermissionError on accessing .guake.yml")
+        content = None
+        try:
+            content = self.fm.read_yaml(filename)
+        except PermissionError:
```

Comment:
These look like exceptions that can always occur with read_yaml(). Maybe these should be inside the function, so that future potential calls don't need to also wrap it in this chunk of exception handling?

Reply:
Sure, there are pleny of possible errors when reading files and expecting the content to have certain format. I just can't think of a good value to return when there is an error with the format, and it seems to me that returning an empty dict would obscure the fact that the file is incorrectly formatted. I could raise a ValueError I guess.

Unless you have a suggestion for a default value to return I would leave it as it is.

Reply:
A None, or a single encapsulating exception, both fine

Reply:
The method `read_yaml` is now returning None in the case of any "expected" error, like PermissionError , NotFoundFile, Encoding or YAMLError. If anything else happens the exception would be capture by the code in `guake_app.py`